commercial/app/controllers/commercial/JobsController.scala

```diff
@@ -22,14 +26,24 @@ class JobsController(jobsAgent: JobsAgent) extends Controller with implicits.Req
     JobSector("media", "Media")
   )
 
+  private def jobSample(specificIds: Seq[String], segment: Segment): Seq[Job] =
+    (jobsAgent.specificJobs(specificIds) ++ jobsAgent.jobsTargetedAt(segment)).distinct.take(2)
+
   def renderJobs = Action { implicit request =>
-    (jobsAgent.specificJobs(specificIds) ++ jobsAgent.jobsTargetedAt(segment)).distinct match {
+    jobSample(specificIds, segment) match {
       case Nil => NoCache(jsonFormat.nilResult.result)
       case jobs => Cached(componentMaxAge) {
         val clickMacro = request.getParameter("clickMacro")
         val omnitureId = request.getParameter("omnitureId")
-        jsonFormat.result(views.html.jobs.jobs(jobs.take(2), jobSectors, omnitureId, clickMacro))
+        jsonFormat.result(views.html.jobs.jobs(jobs, jobSectors, omnitureId, clickMacro))
       }
     }
   }
+
+  def getJobs = Action { implicit request =>
+      val json = Json.toJson(jobSample(specificIds, segment))
+      Cached(60.seconds){
+        JsonComponent(json)
```

Comment:
You can now use the thuper thexy [`JsonComponent(jobSample(specificIds, segment))`](https://github.com/guardian/frontend/blob/master/common/app/common/JsonComponent.scala#L17) (merged this morning). It now has a generic constructor so, as long as it has an implicit `Writes` object, you can pass any old thing in.

This saves you converting it to Json yourself, I know, so much effort.